### PR TITLE
Add support for UTM paramter tracking

### PR DESCRIPTION
**Pull request by grantholle**
This adds support for tracking UTM parameters. I've added the details to the built-in dashboard, but not Nova. I think it would make sense for a Partition metric, but haven't decided 100%.

### CODE REVIEW

**Comment by andreaselia** [comment]
Thanks for your PR @grantholle, much appreciated 👍 

@tomirons what do you think of this? Could it be worth having a `utm_data` json column instead or do you feel like the current column setup is fine?

**Comment by grantholle** [comment]
I had the same thought. I thought this was the simpler/easier approach, as the utm standard seems to be pretty stable and unlikely to change. Happy to go either way

**Comment by tomirons** [comment]
I like the way it is now with the separate columns. 👍

File: tests/Feature/AnalyticsTest.php [hunk 1]
```diff
@@ -129,4 +129,59 @@ public function a_page_view_from_an_excluded_ip_is_not_tracked_if_enabled()
...
+            'utm_campaign' => 'test-campaign',
+            'utm_term' => 'test-term',
+            'utm_content' => 'test-content',
+        ]);
+    }
+
+    /** @test */
+    public function utm_details_will_be_trimmed()
+    {
+        $string = 'Nv19I4yx6b8OEjFhtSqYGANwRim0WJHTQoUDaK3vc72Xl5rMZk1PpFLBdVuCfZgs7TwMeRhxYi9n6CLt2pzqOBGjXSaHwvyJ8KEVNv19I4yx6b8OEjFhtSqYGANwRim0WJHTQoUDaK3vc72Xl5rMZk1PpFLBdVuCfZgs7TwMeRhxYi9n6CLt2pzqOBGjXSaHwvyJ8KEVNv19I4yx6b8OEjFhtSqYGANwRim0WJHTQoUDaK3vc72Xl5rMZk1PpFLBdVuCfZgs7TwMeRhxYi9n6CLt2pzqOBGjXSaHwvyJ8KEV';
```
Comment (andreaselia) [inline tests/Feature/AnalyticsTest.php:165]: Any chance we could use a helper function or something to generate this string or does it need to be this specific one for any specific reason?

**Review by andreaselia — CHANGES_REQUESTED** [review]

File: tests/Feature/AnalyticsTest.php (hunk 1 above)
Comment (grantholle) [reply tests/Feature/AnalyticsTest.php]: No reason in particular. I've updated it to use the string helper

**Pull request merged**